### ISSUE TITLE
Removing route_map_out from a fortios_router_bgp neighbor has no effect

### ISSUE THREAD

**Issue by ebarrett-Ocient** [fortinetdev/terraform-provider-fortios#332]
If `route_map_out` is configured under a `neighbor` within the `fortios_router_bgp` resource and subsequently removed, executing `terraform apply` does not reflect this change.

Steps to Reproduce:

1. Use the following Terraform configuration:

```
resource "fortios_router_bgp" "bgp_config" {
  as          = "123"
  router_id   = "10.10.10.10"
  neighbor {
    ip              = "10.30.2.2"
    remote_as       = 398300
    route_map_out   = "neighbor-out"
  }
}

resource "fortios_router_prefixlist" "neighbor_out" {
  name = "neighbor-out"
  rule {
    id     = "100"
    prefix = "deny"
  }
}
```

2. Run terraform apply.

Remove the line `route_map_out = "neighbor-out"` from the configuration:

```
resource "fortios_router_bgp" "bgp_config" {
  as          = "123"
  router_id   = "10.10.10.10"
  neighbor {
    ip              = "10.30.2.2"
    remote_as       = 398300
  }
}

resource "fortios_router_prefixlist" "neighbor_out" {
  name = "neighbor-out"
  rule {
    id     = "100"
    prefix = "deny"
  }
}
```
4. Run terraform apply again.

Expected Behavior:
Terraform should detect the removal of `route_map_out` and apply the change accordingly.

Actual Behavior:
Terraform finds no changes to apply, despite the removal of `route_map_out` from the configuration.

I have experienced this issue on versions 1.16.0 and 1.20.0.

**Comment by MaxxLiu22** [comment]
Hi @ebarrett-Ocient ,

Thank you for bringing this issue to our attention. Currently, Terraform FOS does not support the unset function, such as unset route_map_out in CLI. However, I have reported this issue to the development team, and they are considering adding this feature. Sorry for any inconvenience.

Thanks,
Maxx

**Comment by ebarrett-Ocient** [comment]
Thanks for the info. Would I have the same issue if I used `fortios_routerbgp_neighbor` instead of defining my neighbors under `fortios_router_bgp`?

**Comment by MaxxLiu22** [comment]
Hi @ebarrett-Ocient ,

I believe that the issue also occurs when using ``fortios_routerbgp_neighbor``. If it is feasible in your environment, you might consider deleting the ``fortios_routerbgp_neighbor`` and then recreating it to achieve the unset function for now. Sorry for the inconvenience.

Thanks,
Maxx

**Comment by misastovicek** [comment]
I think this is the case at multiple places. For example if you add a rule before another rule to a route map and the new rule does not have a property set which has been set on the former one, you'll end up with a wrong rule configuration. It would be nice if the provider called unset in appropriate places in case the property is not defined or is set to null.

**Comment by MaxxLiu22** [comment]
Hi @misastovicek ,

Thank you for your valuable suggestion. We’re pleased to inform you that the development team has included the unset function in our Terraform provider for the upcoming release. If an argument does not have a default value and can be set to empty, Terraform will handle it accordingly.

Thanks,
Maxx

**Comment by MaxxLiu22** [comment]
Hi @zippanto,

I wanted to let you know that this issue has been resolved in Terraform FOS 1.21.0. Starting with this version, Terraform now supports the unset function if the argument is not set. Whenever you have a chance, please upgrade to this version, and feel free to reach out if you have any further questions.

Thank you,
Maxx
